### PR TITLE
Add Support for Cache Size Limits for kopia server start

### PR DESCRIPTION
**Pull request by r4rajat**
## Change Overview

Add Support for Cache Size Limits for command `kopia server start`
This PR will enable support for following flags for `kopia server start` command
- `--cache-directory`
- `--content-cache-size-limit-mb`
- `--metadata-cache-size-limit-mb`

without these flags kopia server sometimes takes a lot of storage for caching in cases when we have large data to backup or restore.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :building_construction: Build

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kopia/command/server_test.go [hunk 1]
```diff
@@ -41,6 +45,8 @@ func (kServer *KopiaServerTestSuite) TestServerCommands(c *C) {
 			f: func() []string {
 				args := ServerStartCommandArgs{
 					CommandArgs:          commandArgs,
+					CacheArgs:            cacheArgs,
+					CacheDirectory:       "cache/dir",
```
Comment (viveksinghggits) [inline pkg/kopia/command/server_test.go:49]: we are passing values of these fields, same everywhere. We should change that in every test case with diff possible values.

**Review by viveksinghggits — COMMENTED** [review]
LGTM

File: pkg/kopia/command/server_test.go (hunk 1 above)
Comment (r4rajat) [reply pkg/kopia/command/server_test.go:49]: @viveksinghggits But on other hand, we're also keeping the `commandArgs` same for all tests, which contains details like `RepoPassword`, `ConfigFilePath` & `LogDirectory`. So thats's why I also kept the `cacheArgs` same for all tests. But if you want we can keep them different. WDYT ?

Comment (viveksinghggits) [reply pkg/kopia/command/server_test.go:49]: I think it should be ok, even if we kept them the same.

File: pkg/kopia/command/server.go [hunk 2]
```diff
@@ -60,6 +62,8 @@ func ServerStart(cmdArgs ServerStartCommandArgs) []string {
 		args = args.AppendLoggable(readOnlyFlag)
 	}
 
+	args = cmdArgs.kopiaCacheArgs(args, cmdArgs.CacheDirectory)
+
```
Comment (PrasadG193) [inline pkg/kopia/command/server.go:66]: [nit] Let's move this after `args := commonArgs()` on line no 41.

**Review by PrasadG193 — APPROVED** [review]
LGTM

File: pkg/kopia/command/server.go (hunk 2 above)
Comment (r4rajat) [reply pkg/kopia/command/server.go]: Done !

**Comment by PrasadG193** [comment]
@r4rajat can you take a look at CI failure?

**Pull request merged**